astropy/time/formats.py

```diff
@@ -2175,6 +2176,204 @@ def value(self):
         return self.mask_if_needed(iterator.operands[-1])
 
 
+class TimeDeltaQuantityString(TimeDeltaFormat, TimeUnique):
+    """Time delta as a string with one or more Quantity components.
+
+    This format provides a human-readable multi-scale string representation of a time
+    delta. It is convenient for applications like a configuration file or a command line
+    option. It is NOT intended for high-precision applications since the internal
+    calculations are done using 64-bit floating point numbers.
+
+    The format is specified as follows:
+
+    - The string is a sequence of one or more components.
+    - Each component is a number followed by an astropy unit of time.
+    - For input, whitespace within the string is allowed but optional.
+    - For output, there is a single space between components.
+    - The allowed components are listed below.
+    - The order (yr, d, hr, min, s) is fixed but individual components are optional.
+
+    The allowed component units are shown below:
+
+    - "yr": years (365.25 days)
+    - "d": days (24 hours)
+    - "hr": hours (60 minutes)
+    - "min": minutes (60 seconds)
+    - "s": seconds
+
+    .. Note:: These definitions correspond to physical units of time and are NOT
+       calendar date intervals. Thus adding "1yr" to "2000-01-01 00:00:00" will give
+       "2000-12-31 06:00:00" instead of "2001-01-01 00:00:00".
+
+    The ``out_subfmt`` attribute specifies the components to be included in the string
+    output.  The default is ``"multi"`` which includes all non-zero components::
+
+    - "multi": multiple components, e.g. "1yr 2d 3hr 5.6s"
+    - "yr": years
+    - "d": days
+    - "hr": hours
+    - "min": minutes
+    - "s": seconds
+
+    Examples
+    --------
+    >>> from astropy.time import Time, TimeDelta
+    >>> import astropy.units as u
+
+    >>> print(TimeDelta("1yr"))
+    1yr
+
+    >>> print(Time("2000-01-01") + TimeDelta("1yr"))
+    2000-12-31 06:00:00.000
+    >>> print(TimeDelta("+1yr 3.6d"))
+    1yr 3d 14hr 24min
+    >>> print(TimeDelta("-1yr 3.6d"))
+    -1yr 3d 14hr 24min
+    >>> print(TimeDelta("1yr 3.6d", out_subfmt="d"))
+    368.85d
+
+    >>> td = TimeDelta(40 * u.hr)
+    >>> print(td.to_value(format="quantity_str"))
+    1d 16hr
+    >>> print(td.to_value(format="quantity_str", subfmt="d"))
+    1.667d
+    >>> td.precision = 9
+    >>> print(td.to_value(format="quantity_str", subfmt="d"))
+    1.666666667d
+    """
+
+    name = "quantity_str"
+
+    subfmts = (
+        ("multi", None, None),
```

Comment:
I still do not like having years included by default - just too confusing - this is going to cause misery!

How about having "dhms" and "ydhms" options? (where I'd put "dhms" first on purpose, i.e., one has to do more work to get the years).

Reply:
Resolved by opening [#15482](https://github.com/astropy/astropy/issues/15482)